test/integration/targets/docker_config/tasks/test_docker_config.yml

```diff
@@ -0,0 +1,102 @@
+- name: Make sure we're not already using Docker swarm
+  docker_swarm:
+    state: absent
+    force: true
+
+- name: Create a Swarm cluster
+  docker_swarm:
+    state: present
+    advertise_addr: "{{ansible_default_ipv4.address}}"
+
+- name: Parameter name should be required
+  docker_config:
+    state: present
+  ignore_errors: yes
+  register: output
+
+- name: assert failure when called with no name
+  assert:
+    that:
+       - 'output.failed'
+       - 'output.msg == "missing required arguments: name"'
+
+- name: Test parameters
+  docker_config:
+    name: foo
+    state: present
+  ignore_errors: yes
+  register: output
+
+- name: assert failure when called with no data
+  assert:
+    that:
+       - 'output.failed'
+       - 'output.msg == "state is present but all of the following are missing: data"'
+
+- name: Create config
+  docker_config:
+    name: db_password
+    data: opensesame!
+    state: present
+  register: output
+
+- name: Create variable config_id
+  set_fact:
+    config_id: "{{ output.config_id }}"
+
+- name: Inspect config
+  command: "docker config inspect {{ config_id }}"
+  register: inspect
+
+- debug: var=inspect
+
+- name: assert config creation succeeded
+  assert:
+    that:
+       - "'db_password' in inspect.stdout"
+       - "'ansible_key' in inspect.stdout"
+
+- name: Create config again
+  docker_config:
+    name: db_password
+    data: opensesame!
+    state: present
+  register: output
+
+- name: assert create config is idempotent
+  assert:
+    that:
+       - not output.changed
+
+- name: Update config
+  docker_config:
+    name: db_password
+    data: newpassword!
+    state: present
+  register: output
+
+- name: assert config was updated
+  assert:
+    that:
+       - output.changed
+       - output.config_id != config_id
+
+- name: Remove config
+  docker_config:
+    name: db_password
+    state: absent
+
+- name: Check that config is removed
+  command: "docker config inspect {{ config_id }}"
+  register: output
+  ignore_errors: yes
+
+- name: assert config was removed
+  assert:
+    that:
+      - output.failed
```

Comment:
You could also add another test which checks that `docker_config` with `state == absent` is idempotent.

Reply:
@felixfontein Sorry, but I don't get this one. I'm not sure under what circumstance could `state == absent` be non-idempotent.

Reply:
It could be that it always returns `changed == True`, for example because deleting doesn't actually work (so it tries to do it both times and fails silently), or because it doesn't detect that it is deleted (and reports that it deleted it another time). Simply trying to delete again and see that the second `state == absent` operation has no change should prevent this.